client/src/components/overview/ImageGalleryModel.js

```diff
@@ -0,0 +1,54 @@
+class ImageGalleryModel {
```

Comment:
Your level of comfort with these layers and layers of abstraction is astonishing to me. No doubt this is problematic, but I'm not qualified to determine whether the problem extends beyond my own lack of experience.

Reply:
Although experience helps to start out more abstract earlier (e.g. learn to code with generics C#! Especially in interfaces!), some of it also is a funny process to see unfold from refactoring. 

Seeing everything all together is easier to read in one pass, but with issues (repeated code, extra code, coupling, etc.). Looking at it all together and grouping it in a certain way, you can better see how to extract related functionality out for making easier code to test, re-use, and maintain... but at the cost of being as obvious everywhere. Although hopefully it improves comprehension at the lower levels of abstraction, like the React files :-)